src/var/distances.jl

```diff
@@ -130,6 +134,92 @@ function distance{T<:MutationType,A<:NucleotideAlphabet}(::Type{Count{T}}, seqs:
 end
 
 """
+    distance{T<:MutationType,A<:NucleotideAlphabet}(::Type{Count{T}}, seqs::Vector{BioSequence{A}}, width::Int, step::Int)
+
+A distance method which computes pairwise distances using a sliding window.
+
+As the window of `width` base pairs in size moves across a pair of sequences it
+computes the distance between the two sequences in that window.
+
+This method computes mutation counts for every window, and returns a tuple of the
+matrix of p-distances for every window, a matrix of the number of valid sites
+counted by the function for each window.
+"""
+function distance{T<:MutationType,A<:NucleotideAlphabet}(::Type{Count{T}}, seqs::Vector{BioSequence{A}}, width::Int, step::Int)
+    mutationFlags, ambiguousFlags = flagmutations(T, seqs)
+    nbases, npairs = size(mutationFlags)
+    @assert width >= 1 "Window width must be ≥ 1."
+    @assert step >= 1 "step must be ≥ 1."
+    @assert width <= nbases "The window size cannot be greater than number of data elements."
```

Comment:
`@assert` should not be used for validating arguments. `ArgumentError` is the dedicated exception type for that purpose.